website/docs/guides/old-runner-migration.md

```diff
@@ -0,0 +1,381 @@
+      the [shebang header](#example-shebang-script-with-scala-cli) defined. This is sort of the equivalent of the
+      old `-howtorun guess`.
+    - To run the `REPL`, refer to [the `repl` sub-command](../commands/repl.md)
+    - This option has been largely replaced with Scala CLI's [sub-commands](../commands/basics.md)
+- `-I`, to preload the extra files for the `REPL`, try passing them as inputs
+  for [the repl sub-command](../commands/repl.md)
+- `-nc` / `nocompdaemon`, the underlying script runner class can no longer be picked explicitly, as with the old `scala`
+  runner
+- `-run` - Scala CLI does not support explicitly forcing the old run mode. Just pass your sources as inputs and ensure
+  they are in the correct format and extension.
+
+### Scala compiler options
+
+All compiler options are supported when passed with the `--scalac-option` flag (or the `-O` alias for short).
+However, many compiler options can also be passed directly.
+For more information, refer
+to [the Scala compiler options section of the `compile` sub-command doc](../commands/compile.md#scala-compiler-options).
+
+## How does Scala CLI detect if it's running a script or a main method?
+
+To answer this question, some disambiguation is necessary.
+The most important thing to note is that this has been handled differently by the 2 old `scala` runners (for
+Scala `2.13.x` and for `3.x`), so a
+consistent behaviour hasn't really been established before Scala CLI.
+
+The Scala `2.13.x` old `scala` runner was the most flexible, automatically detecting if what is being run is a script or
+an
+object based on the source contents. This automatic detection was also possible to be overridden with the `-howtorun`
+runner option (which has been
+deprecated and is not supported in Scala CLI,
+as [noted in an earlier section](#deprecated-and-unsupported-old-scala-runner-options)).
+This also means that the `2.13.x` old `scala` runner did not really care about file extensions much.
+
+In contrast, the Scala `3.x` old `scala` runner relied on file extensions, with some extra consideration for the
+presence of a
+shebang header. Additionally, the Scala `3.x` old runner also inspected for main method definitions in `.sc` files,
+including those
+using [the Scala 3 idiomatic `@main` annotation](https://docs.scala-lang.org/scala3/book/methods-main-methods.html).
+This means that the Scala `3.x` runner respected main methods defined in `.sc` files, but not scripts in `.scala` files.
+
+Scala CLI's approach is perhaps the most restrictive here.
+It accepts explicitly defined main methods in `.scala` sources and script syntax in `.sc` sources, without any
+additional flexibility.
+The only exception would be files with a shebang header ran with the `shebang` sub-command, which are treated as
+scripts (more details about this can be found [in [the shebang scripts' guide](./shebang.md)]).
```

Comment:
Do you mean the files with no file extension or with unsupported one? Because e.g. `.scala` files will still be run as normal sources.